src/main/java/agh/edu/pl/MapNotes/controller/UserController.java

```diff
@@ -0,0 +1,90 @@
+package agh.edu.pl.MapNotes.controller;
+
+import agh.edu.pl.MapNotes.exception.UserNotFoundException;
+import agh.edu.pl.MapNotes.model.User;
+import agh.edu.pl.MapNotes.model.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private final UserRepository userRepository;
+
+    @Autowired
+    UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * Get all users in the database.
+     * @return list of all users in the database.
+     */
+    @GetMapping
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * Update or create map.
+     * Function allow to create user or to update.
+     * If the id match id of any user it will update that user.
+     * Otherwise it will create new user with new id.
+     * @param user map to update or create.
+     * @return map stored in database after operation.
+     */
+    @PutMapping
+    public User putUser(@Valid @RequestBody User user) {
+        user = userRepository.save(user);
+        return this.userRepository.findById(user.getId()).get();
+    }
+
+
+    @RequestMapping("/addIfNotExist")
+    public Boolean putUserIfNotInDatabase(@RequestBody User user) {
```

Comment:
Wydaje mi się że chodzi tutaj o zwykła funkcje register user. W takim wypadku zwracał bym nowego użytkownika bo jes tam jego id, a sama informacja true/false nam wiele nie mówi. Ponad to zamiast false lepiej podać wyjątek z informacją co jest nie tak.